packages/remix-cloudflare-pages/worker.ts

```diff
@@ -0,0 +1,96 @@
+import type { ServerBuild, AppLoadContext } from "@remix-run/server-runtime";
+import { createRequestHandler as createRemixRequestHandler } from "@remix-run/server-runtime";
+
+export interface CreateRequestHandlerParams<Env = any> {
+  build: ServerBuild;
+  getLoadContext?: (context: EventContext<Env, any, any>) => AppLoadContext;
+  mode?: string;
+}
+
+function createInternalHandler<Env = any>({
+  build,
+  getLoadContext,
+  mode
+}: CreateRequestHandlerParams<Env>): PagesFunction<Env> {
+  let platform = {};
+  let handleRequest = createRemixRequestHandler(build, platform, mode);
+
+  return context => {
+    let loadContext =
+      typeof getLoadContext === "function"
+        ? getLoadContext(context)
+        : undefined;
+    return handleRequest(context.request, loadContext);
+  };
+}
+
+type EnvWithAssets = Record<string, unknown> & {
+  ASSETS: { fetch: typeof fetch };
+};
+
+async function handleAsset(request: Request, env: unknown) {
+  let envWithAssets = env as EnvWithAssets;
+  if (!envWithAssets?.ASSETS?.fetch) {
+    throw new Error(
+      "env.ASSETS.fetch does not exist, did you forget to upload your assets?"
+    );
+  }
+
+  const response = await envWithAssets.ASSETS.fetch(request);
+  if (response.ok) return response;
+}
+
+declare const process: any;
+
+export function createRequestHandler<Env = any>({
```

Comment:
I think `createFetchHandler` is a slightly more correct description for this function.

Reply:
Ah, sorry! I didn't realize you were using `functions/[[path]].ts`. `createRequestHandler` is totally fine! I was thinking you were using `public/_worker.js`.